internal/batches/service/service.go

```diff
@@ -28,6 +28,10 @@ type Service struct {
 	client           api.Client
 	features         batches.FeatureFlags
 	imageCache       *docker.ImageCache
+
+	// TODO(mrnugget): I don't like this state here, ugh.
+	exec  executor.Executor
+	cache executor.ExecutionCache
```

Comment:
If we have a separate entity that can do the whole check-cache+build-specs stuff, then we don't need this anymore.